src/utils/src/url.js

```diff
@@ -9,6 +9,7 @@ import { isDefined } from './object';
 
 /**
  * @private
+ * Adapted from https://github.com/lakenen/node-append-query, to fit a non-node runtime.
  */
 function serialize(obj, options = {}, prefix) {
```

Comment:
I only copied the code from `node-append-query`. This is what it was doing but maybe it is possible we can use the `qs` library to serialize as well.